app/src/main/java/io/gnosis/safe/ui/transactions/details/view/TxConfirmationsView.kt

```diff
@@ -0,0 +1,203 @@
+import io.gnosis.safe.ui.settings.view.AddressItem
+import io.gnosis.safe.utils.dpToPx
+import pm.gnosis.model.Solidity
+import pm.gnosis.svalinn.common.utils.getColorCompat
+
+class TxConfirmationsView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0
+) : LinearLayout(context, attrs, defStyleAttr) {
+
+    private val linePaint: Paint
+
+    init {
+        orientation = VERTICAL
+        gravity = Gravity.LEFT
+        linePaint = Paint(Paint.ANTI_ALIAS_FLAG)
+        linePaint.style = Paint.Style.FILL_AND_STROKE
+        linePaint.strokeWidth = dpToPx(LINE_WIDTH).toFloat()
+        linePaint.color = context.getColorCompat(LINE_COLOR)
+    }
+
+    fun setExecutionData(status: TransactionStatus, confirmations: List<Solidity.Address>, threshold: Int, executor: Solidity.Address? = null) {
+
+        clear()
+
+        addExecutionStep(TxExecutionStep.Type.CREATED)
+
+        confirmations.forEach {
+            addExecutionStep(TxExecutionStep.Type.CONFIRMED)
+            addAddressItem(it)
+        }
+
+        if (confirmations.size < threshold) {
+            addExecutionStep(TxExecutionStep.Type.EXECUTE_WAITING, threshold - confirmations.size)
+        } else {
+
+            when (status) {
+                TransactionStatus.AWAITING_EXECUTION -> {
+                    addExecutionStep(TxExecutionStep.Type.EXECUTE_READY)
+                }
+                TransactionStatus.SUCCESS -> {
+                    if(executor != null) {
+                        addExecutionStep(TxExecutionStep.Type.EXECUTE_DONE)
+                        addAddressItem(executor)
+                    } else {
+                        // fail silently
+                        addExecutionStep(TxExecutionStep.Type.EXECUTE_READY)
+                    }
+                }
+                TransactionStatus.CANCELLED -> {
+                    addExecutionStep(TxExecutionStep.Type.CANCELED)
+                }
+                TransactionStatus.FAILED -> {
+                    addExecutionStep(TxExecutionStep.Type.FAILED)
+                }
+            }
+        }
+    }
+
+    private fun addExecutionStep(type: TxExecutionStep.Type, missingConfirmations: Int? = null) {
+        addView(TxExecutionStep(context).apply {
+            val lp = LayoutParams(LayoutParams.WRAP_CONTENT, LayoutParams.WRAP_CONTENT)
+            lp.setMargins(0, 0, 0, dpToPx(MARGIN_VERTICAL))
+            layoutParams = lp
+            setStep(type, missingConfirmations)
+        })
+    }
+    
+    private fun addAddressItem(address: Solidity.Address) {
+        val addressItem = AddressItem(context)
+        val layoutParams = LayoutParams(LayoutParams.MATCH_PARENT, dpToPx(ADDRESS_ITEM_HEIGHT))
+        layoutParams.setMargins(dpToPx(ADDRESS_ITEM_MARGIN_LEFT), 0, 0, dpToPx(MARGIN_VERTICAL))
+        addressItem.layoutParams = layoutParams
+        addressItem.address = address
+        addView(addressItem)
+    }
+
+    fun clear() {
```

Comment:
do you need access from outside? I would either inline this method or make it `private`.

Reply:
this can be potentially used from outside. It's part of public api of the view.